pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagent.go

```diff
@@ -124,7 +125,7 @@ func (h *handler) OnChange(cluster *rancherv1.Cluster, status rancherv1.ClusterS
 	result = append(result, &v1alpha1.Bundle{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: cluster.Namespace,
-			Name:      name.SafeConcatName(cluster.Name, "managed", "system", "agent"),
+			Name:      name.Limit(name.SafeConcatName(cluster.Name, "managed", "system", "agent"), rke2.MaxHelmReleaseNameLength),
```

Comment:
QQ, if we have a cluster name that is >= 53 characters, won't we end up creating identical bundles?

Reply:
@jakefhyde  As in duplicate names for both the SUC and fleet-agent bundle? The additional strings we pass to `SafeConcatName` would ensure those two bundles don't share a name, though I might be misunderstanding your question

Reply:
If it gets truncated to 53 characters, and you can have a 64 character cluster name, then any cluster with the first 53 matching characters of another cluster will fail.

Reply:
ACtually should be fine, looks like `SafeConcatName` and `name.Limit` append the digest of the whole string, not the truncated version.

Reply:
Just to follow up, this PR has been changed to no longer use a combination of wranglers `SafeConcatName` and `name.Limit` as it resulted in a double hashing of the string. A few function was added in Rancher and should be ported into Wrangler when we get the opportunity